src/otg/datasource/finngen/finngen_finemapping.py

```diff
@@ -0,0 +1,234 @@
+            .option("compression", "gzip")
+            .csv(finngen_finemapping_df, header=True)
+            # Drop rows which don't have proper position.
+            .filter(f.col("position").cast(t.IntegerType()).isNotNull())
+            # Drop non credible set SNPs:
+            .filter(f.col("cs") > 0)
+            .select(
+                # Add study idenfitier.
+                f.concat(f.lit(finngen_release_prefix), f.col("trait"))
+                .cast(t.StringType())
+                .alias("studyId"),
+                f.col("region"),
+                # Add variant information.
+                f.regexp_replace(f.col("v"), ":", "_").alias("variantId"),
+                f.col("cs").cast("integer").alias("credibleSetIndex"),
+                f.regexp_replace(f.col("chromosome"), "^chr", "")
+                .cast(t.StringType())
+                .alias("chromosome"),
+                f.col("position").cast(t.IntegerType()),
+                f.col("allele1").cast(t.StringType()).alias("ref"),
+                f.col("allele2").cast(t.StringType()).alias("alt"),
+                # Parse p-value into mantissa and exponent.
+                *parse_pvalue(f.col("p")),
+                # Add beta, standard error, and allele frequency information.
+                f.col("beta").cast("double"),
+                f.col("se").cast("double").alias("standardError"),
+                f.col("maf").cast("float").alias("effectAlleleFrequencyFromSource"),
+                f.lit("SuSie").cast("string").alias("finemappingMethod"),
+                f.col("alpha1").cast("double").alias("alpha_1"),
+                f.col("alpha2").cast("double").alias("alpha_2"),
+                f.col("alpha3").cast("double").alias("alpha_3"),
+                f.col("alpha4").cast("double").alias("alpha_4"),
+                f.col("alpha5").cast("double").alias("alpha_5"),
+                f.col("alpha6").cast("double").alias("alpha_6"),
+                f.col("alpha7").cast("double").alias("alpha_7"),
+                f.col("alpha8").cast("double").alias("alpha_8"),
+                f.col("alpha9").cast("double").alias("alpha_9"),
+                f.col("alpha10").cast("double").alias("alpha_10"),
+                f.col("lbf_variable1").cast("double").alias("lbf_1"),
+                f.col("lbf_variable2").cast("double").alias("lbf_2"),
+                f.col("lbf_variable3").cast("double").alias("lbf_3"),
+                f.col("lbf_variable4").cast("double").alias("lbf_4"),
+                f.col("lbf_variable5").cast("double").alias("lbf_5"),
+                f.col("lbf_variable6").cast("double").alias("lbf_6"),
+                f.col("lbf_variable7").cast("double").alias("lbf_7"),
+                f.col("lbf_variable8").cast("double").alias("lbf_8"),
+                f.col("lbf_variable9").cast("double").alias("lbf_9"),
+                f.col("lbf_variable10").cast("double").alias("lbf_10"),
+            )
+            .withColumn(
```

Comment:
There's a lot of repetition in the extraction of logBF and posteriorProbability. I'd suggest you extract this logic in a separate method.